x-pack/test/api_integration/deployment_agnostic/services/package_api.ts

```diff
@@ -18,50 +17,52 @@ export interface IntegrationDataset {
   type: 'logs' | 'metrics' | 'synthetics' | 'traces';
 }
 
-export function PackageApiProvider({ getService }: DeploymentAgnosticFtrProviderContext) {
-  const samlAuth = getService('samlAuth');
-  const supertestWithoutAuth = getService('supertestWithoutAuth');
-
+export function PackageApiProvider() {
   return {
     async installCustomIntegration({
-      roleAuthc,
+      roleScopedSupertestWithCookieCredentials,
       customIntegration,
     }: {
-      roleAuthc: RoleCredentials;
+      roleScopedSupertestWithCookieCredentials: SupertestWithRoleScopeType;
       customIntegration: CustomIntegration;
     }) {
       const { integrationName, datasets } = customIntegration;
 
-      const { body } = await supertestWithoutAuth
+      const { body } = await roleScopedSupertestWithCookieCredentials
         .post(`/api/fleet/epm/custom_integrations`)
```

Comment:
I think it wasn't intentional: Fleet API is public (no `internal` in route path) and should be called with API key as it was before change.

I think you need to revert it and pass API key from test. You can get it similar to cookie header:

```
      supertestAdminWithApiKey = await roleScopedSupertest.getSupertestWithRoleScope(
        'admin',
        {
          withInternalHeaders: true,
        }
      );
```

Keep in mind to invalidate it in `after` hook

Reply:
I get your point. The reason i did that was for all our tests, which are internal API, i was already generating `roleScopedSupertestWithCookieCredentials`, so i though to use the same for Fleet packages. 

But your point is valid as fleet api is public, i must authenticate using API key rather than header. I will get both API key (for Fleet) and headers for my test